### PR TITLE
Declare plugin deps in postcss_plugin instead of postcss_binary

### LINKED ISSUE #19: Avoid double-specifying plugins
~~If #17 is resolved, we'll start writing this:~~

Currently plugins are double-specified with their defs:

```python
postcss_plugin(
    name = "list_selectors",
    node_require = "build_bazel_rules_postcss/examples/additional_outputs/list-selectors.js",
    srcs = [
        "list-selectors.js",
    ],
)

postcss_binary(
    name = "style_processed",
    src = "style.css",
    plugins = {
        ":list_selectors": "[{markdownDir: 'examples/additional_outputs'}]",
    },
    additional_outputs = ["selectors.md"],
    deps = [
        ":list_selectors",
    ],
)
```

We could do one better by not needing to specify `:list_selectors` in the `deps` attribute:

```python
postcss_plugin(
    name = "list_selectors",
    node_require = "build_bazel_rules_postcss/examples/additional_outputs/list-selectors.js",
    srcs = [
        "list-selectors.js",
    ],
)

postcss_binary(
    name = "style_processed",
    src = "style.css",
    plugins = {
        ":list_selectors": "[{markdownDir: 'examples/additional_outputs'}]",
    },
    additional_outputs = ["selectors.md"],
)
```

This would likewise apply for things we get from npm:

```python
postcss_plugin(
    name = "autoprefixer",
    node_require = "autoprefixer",
)

postcss_plugin(
    name = "unquote",
    node_require = "build_bazel_rules_postcss/examples/custom_plugin/unquote.js",
    srcs = [
        "unquote.js",
    ],
)

sass_binary(
    name = "style",
    src = "style.scss",
    output_style = "expanded",
)

postcss_binary(
    name = "style_processed",
    src = ":style",
    plugins = {
        ":autoprefixer": "[{ browsers: '%s' }]" % AUTO_PREFIXER_BROWSERS,
        ":unquote": "",
    },
    deps = [
        ":unquote",
        "@npm//autoprefixer",
    ],
)
```

Would become something like this:

```python
postcss_plugin(
    name = "autoprefixer",
    node_require = "autoprefixer",
    deps = [
        "@npm//autoprefixer",
    ],
)

postcss_plugin(
    name = "unquote",
    node_require = "build_bazel_rules_postcss/examples/custom_plugin/unquote.js",
    srcs = [
        "unquote.js",
    ],
)

sass_binary(
    name = "style",
    src = "style.scss",
    output_style = "expanded",
)

postcss_binary(
    name = "style_processed",
    src = ":style",
    plugins = {
        ":autoprefixer": "[{ browsers: '%s' }]" % AUTO_PREFIXER_BROWSERS,
        ":unquote": "",
    },
)
```

### PR DESCRIPTION
**Pull request by rzhw**
Resolves #19

### CODE REVIEW

File: internal/binary.bzl [hunk 1]
```diff
@@ -50,11 +50,10 @@ def postcss_binary(
             * `bazel.additionalOutputs`: An array of all the file paths passed
               to the `additional_outputs` attribute of `postcss_binary`, to
               which a plugin is expected to write.
-
-        deps: A list of NodeJS modules the config depends on. The PostCSS module
-            is always implicitly included.
         src: The input .css, and optionally .css.map files. (This includes
             outputs from preprocessors such as sass_binary.)
+        deps: A list of NodeJS modules the config depends on. The PostCSS module
+            is always implicitly included.
```
Comment (nex3) [inline internal/binary.bzl]: It's probably worth mentioning that all plugin modules are implicitly included.

File: internal/binary.bzl [hunk 2]
```diff
@@ -26,8 +26,8 @@ load(":run.bzl", "postcss_run")
 def postcss_binary(
```
Comment (nex3) [inline internal/binary.bzl:26]: This still says that plugins' keys are require paths, but now they're deps.

File: internal/plugin.bzl [hunk 3]
```diff
@@ -19,10 +19,26 @@ Node.js source files, as well as promoting reuse of plugins across multiple
...
 PostcssPluginInfo = provider(fields=["node_require"])
 
 def _postcss_plugin_info_impl(ctx):
-    return [PostcssPluginInfo(node_require=ctx.attr.node_require)]
+    direct_sources = depset(ctx.files.srcs)
+    sources_depsets = [direct_sources]
+
+    for dep in ctx.attr.deps:
+        if NpmPackageInfo in dep:
+            sources_depsets.append(dep[NpmPackageInfo].sources)
```
Comment (nex3) [inline internal/plugin.bzl]:
```suggestion
    sources = depset(
        ctx.files.srcs,
        transitive = [
            dep[NpmPackageInfo].sources
            for dep in ctx.attr.deps
            if NpmPackageInfo in dep
        ]
    )
```

**Review by nex3 — APPROVED** [review]

File: internal/binary.bzl (hunk 1 above)
Comment (rzhw) [reply internal/binary.bzl]: Done.

File: internal/binary.bzl (hunk 2 above)
Comment (rzhw) [reply internal/binary.bzl:26]: Done.

File: internal/plugin.bzl (hunk 3 above)
Comment (rzhw) [reply internal/plugin.bzl]: Done.

Adjusted whitespace as per http://google.github.io/styleguide/pyguide.html#274-decision

**Pull request merged**